src/sys/time.rs

```diff
@@ -147,32 +147,40 @@ impl ops::Div<i32> for TimeVal {
 
 impl fmt::Display for TimeVal {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let (abs, sign) = if self.tv_sec < 0 {
+        let (abs, sign) = if self.0.tv_sec < 0 {
             (-*self, "-")
         } else {
             (*self, "")
         };
 
-        let sec = abs.tv_sec;
+        let sec = abs.0.tv_sec;
 
         try!(write!(f, "{}", sign));
 
-        if abs.tv_usec == 0 {
-            if abs.tv_sec == 1 {
+        if abs.0.tv_usec == 0 {
+            if abs.0.tv_sec == 1 {
                 try!(write!(f, "{} second", sec));
             } else {
                 try!(write!(f, "{} seconds", sec));
             }
-        } else if abs.tv_usec % 1000 == 0 {
-            try!(write!(f, "{}.{:03} seconds", sec, abs.tv_usec / 1000));
+        } else if abs.0.tv_usec % 1_000 == 0 {
+            try!(write!(f, "{}.{:03} seconds", sec, abs.0.tv_usec / 1_000));
         } else {
-            try!(write!(f, "{}.{:06} seconds", sec, abs.tv_usec));
+            try!(write!(f, "{}.{:06} seconds", sec, abs.0.tv_usec));
         }
 
         Ok(())
     }
 }
 
+impl PartialEq for TimeVal {
+    fn eq(&self, rhs: &TimeVal) -> bool {
+        self.0.tv_sec == rhs.0.tv_sec && self.0.tv_usec == rhs.0.tv_usec
+    }
+}
+
+impl Eq for TimeVal { }
+
```

Comment:
could you also implement `PartialOrd`, `Ord`, and `Debug`, which we lost in move to libc. I figure that `Debug` could print like `TimeVal(timeval { tv_sec: 10, tv_usec: 20 })` as the derived impl would have. I copied the definitions to playground to demonstrate what `derive(Debug)` would do: https://is.gd/jK1j9K